Auto-Spunding

MrDAnger82 opened the pull request:
Hello.
Assembled the BPL and assembled an automatic tongue for pressure control. I use WemosMini and in order not to burn A0 between the pressure sensor I put a 4.7k resistor and calibrated at 4.4psi. When calibrating the sensor, the parameter "b" varies 137-155 for me, and the parameter "a" 0.32 ..- 0.48 ...
Pressure control works, but the readings are very "floating", for example, then 17psi, then 25psi, then 46psi. If the control is set to 30psi, the machine relieves the pressure and in the end I get about 15-18psi. Checked when supplying gas from a cylinder.
Maybe it makes sense to add pressure logging and its graph to the screen?
Then it will be possible to deduce approximately how much pressure should be indicated above in the control to obtain the desired carbonation of beer.
The sensor is at 500psi, just like what you have in the picture, only bought to China on Aliexpress.

vitotai commented:
First of all, I am glad that it works. I haven't finished hardware upgrade to my fermenting controller.

You might have a pressure transducer with TOO high maximum PSI and hence error. The specification of mine is 100psi max, and I got around 3psi variation, which is exact 3% error specified. If you have the same one as mine, and I bet you do, maximum 15psi error seems reasonable.

You must have as small as possible "pin hole" for pressure release. It is very important, although I guess you should have already done a good job. Given no interrupt is available, pulling is the only way I can think of. ( I've thought of using another controller dedicated for pressure control, which complicated things a little bit.) To avoid exhausting the processor, the pressure is checked every second. That means the valve will be open for at least one second before the controller has a chance to check the pressure reading. 

Some kind of filter is necessary. Maybe a simple low pass filter might do. Or, 5 successive reads from A0, and average the middle 3 readings. What is your opinion?

Adding pressure readings to chart is in the schedule, after I finish my setup for better verifying.

Specifying carbonation level is a good idea. However, I have to study/survey the formula first. A chart is the only way I know.

MrDAnger82 commented:
Today, the tongue apparatus has been slightly modified: I removed the resistor altogether. Because the maximum pressure is unlikely to exceed 44psi, so I doubt that a large voltage will come to the input of the processor.
Everything is working.
The power supply apparently also has a range of parameters. I had a Chinese 5v 1A unit, replaced it with the original Samsung 5v 2A power supply, the readings began to jump less.
I also replaced the wire from the sensor with a shielded one (USB-wire), and applied grounding to the braid.
Again, the readings are more stable. But the problem remains ...
I have a sensor from 0psi to 500psi, during calibration I apply pressure no more than 44psi.

MrDAnger82 commented:
I am very confused by the jumps of indicators, and they can be negative (-15psi) and sharply positive (112-162psi).
positives rarely occur, but they show a significant effect on the retention of pressure in the system.
After the sensor, I put a needle valve in front of the relief valve, which allows you to adjust the pressure relief rate.
I think that a filter is needed from sharply incorrect results, since there can be no negative pressure, and there can also be no jumps from 30-40 psi to 112-160 psi during fermentation.
Maybe it’s worth checking the pressure 1-2 times per second, taking 2 close results and evaluating them — is it necessary to open the valve?

MrDAnger82 commented:
Yes, and I did not warn, I am from Russia, so my English = English from Google translator ... sorry ...

By the way, we mainly use different units of pressure measurement, for older brewers, usually everything is measured in bar ...
Therefore, I would like to have a choice of pressure indication units ...
But I almost adapted to psi ...

vitotai commented:
I added the filter in a new testing branch. You can try it.
https://github.com/vitotai/BrewPiLess/tree/PressureMonitor

However, I don't think it would work well with variation like that.

MrDAnger82 commented:
Maybe this resource can be useful for creating and editing an automatic carbonization module? 
http://www.lowoxygenbrewing.com/brewing-methods/spunding-automation-and-logic/

MrDAnger82 commented:
> I added the filter in a new testing branch. You can try it.
> https://github.com/vitotai/BrewPiLess/tree/PressureMonitor
> 
> However, I don't think it would work well with variation like that.

Thanks you. I'll try, according to the results, I will write to you.

MrDAnger82 commented:
Vito, the processor creates a BPL network, I try to connect, it says that the connection fails. Authentication to the BPL network does not pass from a phone or tablet.
I double-checked this firmware on several processors, old versions of the firmware on the same processors easily raise the network and everything works. This firmware does not raise. We recompiled the firmware several times, changed the processors, and everything that can be erased from them ... Does not help.

MrDAnger82 commented:
Vito, do not consider it arrogance, but you can’t compile me a bin file? In order to avoid my mistakes?

vitotai commented:
Get the binary with 3.6p.

vitotai commented:
![螢幕快照 2019-08-27 下午3 15 41](https://user-images.githubusercontent.com/15953213/63749547-422c3880-c8de-11e9-94d8-0fc46e394122.jpg)

FYI,
I don't get stable readings from my sensor, either. The correct reading should be steady 0. The 1.8 peak value is already filtered. It's 5psi raw.

MrDAnger82 commented:
Vito, flashed your firmware to several controllers. I can not connect to the brewpiless network. It seems that the controller does not turn on the network ...
Old firmware-without a problem create a network ...
Is there a problem with this new belief: "ESP8266 won't start after selecting WiFi network. The web server used is ESPAsyncWebServer which uses ESPAsyncTCP. I found that if ESP8266 ever enters SoftAP mode before connecting to the network, the Web server will fail on tcp_bind () and the web service will be unavailable. Not tracing the source code of the LWIP, I just worked around by reseting the system. However, ESP8266 sometimes doesn't reset. "

vitotai commented:
Well. I've changed the design. Now, there is no WiFiManager. 
I guess it might have something to do with settings. 
Try flashing via USB. If it still doesn't work. Try turning serial debug, ie. development build, and check the serial logs.

MrDAnger82 commented:
Vito, I'm just a doctor ... :(
I can not do that... :(
I'm flashing through a USB program
ESP8266FlasherX64.

MrDAnger82 commented:
And you can’t stick a filter in the old firmware?

MrDAnger82 commented:
Vito, I managed to start, first I flashed version 3.5.4, then tied it to my home network, and then updated it to version 3.6.4.
Indications are quite stable, they do not jump much.
But I don’t see the pressure graph, as you mentioned above by type ...
And how do you know exactly which version of the firmware is installed? otherwise only v3.6 is written ...
Thank you so much for the device!

MrDAnger82 commented:
Maybe it makes sense to do a trickier sensor calibration? Not by two points, but more? By type, how fast is the iSpindel calibration?

vitotai commented:
The graph is currently under development. 
The calibration processes of iSpindel and pressure transducer are different.
The real formula of iSpindel is an arctangent formula, if we can find one. Linear regression is used to get a proximal formula.
The formula of the transducer is simply a linear equation. It's easy to get the coefficients by two points, or readings.
The problem is not the formula, but the READINGs that the transducer gives are not stable. The program does a very easy job by translate the VOLTAGE reading from A0, the analog input PIN, and convert it to pressure based on the linear formula.
Take mime for example, the reading drifts from 130 to 230 at no pressure, or 0 PSI. It's not possible to get a formula that f(130)=F(155)=f(230)=0. (Except that the formula always return 0 whatever input is.)

MrDAnger82 commented:
It is a pity that the pressure schedule is not available ... I wanted to brew beer at the beginning of September and check its operation ...
I realized that iSpindel has a different dependence ... But its readings also "float" from gas bubbles and waves?
Can measure, say, 5-10 times, just take the arithmetic average? Suddenly it will work?
I also think that we should try to replace the BPL power from a switching power supply with a stabilized transformer. I did not install a Chinese power supply, but the original high-quality from a Samsung tablet, the readings were more stable.
Thank you for your help!

MrDAnger82 commented:
So our problem is the sensor and the analog-to-digital converter of the processor ...
How can I stabilize the data from the sensor yet? Maybe a different connection scheme? With a resistor or capacitor?
Why here:

http://www.lowoxygenbrewing.com/brewing-methods/spunding-automation-and-logic/

Does a person on a similar sensor have no such problems?

vitotai commented:
First of all, you don't know how good it really works.
(well, I think I found some information about his setup. you can check the picture in this thread: https://www.homebrewtalk.com/forum/threads/my-mostly-automated-brucontrol-setup.646559/page-5#post-8389647
The pressure is neither stable.)

Second, I don't see the specification of the sensor. It might be a maximum 35psi one, which would have lower error rate, like 1 or 2 psi.

Third, I got this idea from here:
https://aussiehomebrewer.com/threads/spunding-valve.90341/

He seems to use a 36psi transducer and Arduino. Maybe ADC of ESP8266 might have something to do? I will test it with an Arduino later.

BTW, I found the formula to convert carbonation level into pressure. However, it might not work as expected because It takes some time to reach equilibrium. When crashing the beer, temperature should drop faster than pressure. If the setting pressure is derived from carbonation level, we will lose some co2 before it reach equilibrium. 
I like the idea, though. A simple calculator to help deciding pressure will be added.

vitotai commented:
A little googling led me to this issue:
https://github.com/esp8266/Arduino/issues/2070

I will try me best, but we might not be able to get stable readings unless turning off WiFi, which is non sense to me.

External ADC is an option, like:
https://www.adafruit.com/product/1085
(It is very cheap on eBay.)

MrDAnger82 commented:
Vito, then I'm right in something ...
As I understand it, you need to add a separate voltage stabilizer or stabilized power source for the sensor, the processor ground and the sensor are common.
I will try to do so.
Vito for the experiment would like to have a pressure graph, at least inaccurate, but there will be something to show you.
Thank!

MrDAnger82 commented:
Of course, you should not turn off WiFi, without it there is little sense ...
If it turns out to be bad, can it make sense to make an additional processor (Wemos mini, for example) to work only with the remote sensor, and will it transmit data to the "main" processor?

vitotai commented:
Just push the updated code with graph.
https://github.com/vitotai/BrewPiLess/tree/v37alpha

The tricks mentioned in the issue above are used, and it seems to work fine for me. I got around 0.5 psi maximum drift in 4 hours, which is acceptable.

MrDAnger82 commented:
Updated by OTA ... again I can not connect to the BPL ...
Like before...

MrDAnger82 commented:
The LED constantly blinks.

vitotai commented:
oops. My bad. 
Please try again with the *pre2.bin

MrDAnger82 commented:
Hooray! Everything is working! I'm chasing now. Tomorrow I will try to throw off the screenshots.
Thanks you!

MrDAnger82 commented:
Yes, it got better, but all the same, in an hour it loses pressure to the minimum due to false positives.
It is necessary to try to put a voltage regulator on the sensor power supply or a separate power supply to the common ground.

MrDAnger82 commented:
from 19.30 to 22.20 I threw off the pressure to 5-8PSI with the established 20PSI.
Previous pressure dips - depressurization of the circuit by me.
![ШПУНТ](https://user-images.githubusercontent.com/54216845/64039952-89c3f600-cb64-11e9-9aff-fc3c64015565.png)

lbussy commented:
At this point, if you have to use isolated power, etc., it might be better to use the A/D converter?

MrDAnger82 commented:
Vito, this seems to be a victory.
Connected an additional power source, connected the earth.
Look, a huge difference.
But as before, it drops below 20psi (I set this level), on the control manometer 17psi, and on the sensor 15psi.
But still, the data is stable.
I drive the dowel longer.
I associate the pressure drop with a leak in the path, it will be necessary to do it on tight joints.
![шпунт2](https://user-images.githubusercontent.com/54216845/64045645-7750b900-cb72-11e9-9d94-75c2c7e7aed9.png)

MrDAnger82 commented:
Will there be an automatic maintenance of carbonization by volume depending on temperature?
Suppose I introduce that I need carbonation of 2.61 at a temperature of 6 ° C, and I have a temperature of 7 or 8 — will the BPL maintain the desired level of carbonation?

MrDAnger82 commented:
Yes, the readings are very stable ... Can I try to remove the filter?
![Шпунт3](https://user-images.githubusercontent.com/54216845/64049944-3f9c3e00-cb7f-11e9-8e14-926ff619af4b.png)

MrDAnger82 commented:
Indicator drift within 2-2.5 PSI
![шпкнт4](https://user-images.githubusercontent.com/54216845/64066199-948f9100-cc1f-11e9-9c3e-260ddfd2aaac.png)

MrDAnger82 commented:
> На этом этапе, если вам необходимо использовать изолированное питание и т. Д., Может быть, лучше использовать аналого-цифровой преобразователь?

As a layman, it’s better for me to attach another power supply ... I can do this ... And, here, I can’t do an analog-to-digital converter ...
Moreover, the addition of additional power to the sensor gave good results with minimal effort on my part.

vitotai commented:
> Yes, the readings are very stable ... Can I try to remove the filter?
> ![Шпунт3](https://user-images.githubusercontent.com/54216845/64049944-3f9c3e00-cb7f-11e9-8e14-926ff619af4b.png)

Try binaries with v3.7preNF.

 
> At this point, if you have to use isolated power, etc., it might be better to use the A/D converter?

Sure. It worths a try. 

> Will there be an automatic maintenance of carbonization by volume depending on temperature?
> Suppose I introduce that I need carbonation of 2.61 at a temperature of 6 ° C, and I have a temperature of 7 or 8 — will the BPL maintain the desired level of carbonation?

I've thought of this and answered it in previous comment. The short answer is that it won't work as expected. It takes some time to reach pressure equilibrium. When crashing the beer, temperature should drop faster than pressure. If the setting pressure is derived from carbonation level, we will lose some co2 before it reach equilibrium.

MrDAnger82 commented:
I tried ... No, I need a filter. Take-off run at 7.5PSI at atmospheric pressure.
The filter still has to filter out negative values ​​- in our case, the thing is elusive ... Therefore, everything below 0 should return 0.
With the filter working, the run-up of the readings was 2-2.5psi, which corresponds to a 100mm mercury column - insignificant, IMHO, fluctuations ... But maybe I'm wrong ...
![шпунт6](https://user-images.githubusercontent.com/54216845/64081715-86b83980-cd0d-11e9-92a3-bfc35efb7104.png)

vitotai commented:
The price of low pass filter is latency. It should be about 6~7 seconds latency.
Yet, current implementation seems to be unstable. The reason might be disabling the interrupt during read of ADC.

I have ordered a ADS1115 and will implement external ADC. You might order one now if you want to try.

MrDAnger82 commented:
I do not really like the idea of ​​an extra add-on in the form of an ADC ... But since you decided so, I have no other choice, apparently ... In what form is this ADC ordered? as a ready-made module for Arduino? Nevertheless, I will try the work of the tongue in real beer (Clone Urquell), and there the module will come from China. 
Can I have an offtopic question? How to implement separate BPL control of fermenting beer temperature and glycol temperature control in a glycol cooler? I want to remove the "extra" temperature controller from the glycol cooler. Beer temperature and density are measured by iSpidel.

Is such an ADC suitable?
https://ru.aliexpress.com/item/32716184973.html?spm=a2g0o.productlist.0.0.5db62e35ruuGTY&algo_pvid=dcca9fd5-f690-413b-ba1c-176351b4e94c&algo_expid=dcca9fd5-f690-413b-ba1c-176351b4e94c-0&btsid=f4c32b81-33cf-46f1-9254-0c0216f9673f&ws_ab_test=searchweb0_0,searchweb201602_4,searchweb201603_52

vitotai commented:
There will be an option to choose between internal ADC or external ADS1115. The reason I choose to try ADS1115 is that I found that disabling interrupt makes the system unstable. Also the latency introduced by the filter will result in inaccurate pressure control. Internal ADC is fine if rough readings are needed, but it's not when control is needed.

The ADC is good. I bought mime from eBay, and it is from China also.


I am not sure what you want to do with glycol.
The way to use BPL with glycol as chilling is described in this document:
https://github.com/vitotai/BrewPiLess/blob/master/doc/glycol.md

The way to use iSpindel as beer sensor is described here:
https://github.com/vitotai/BrewPiLess/blob/master/doc/iSpindelAsSensor.md

If you want BPL to control the temperature of glycol, you will need to setup a "control PIN/actuator" for it. Then you will see a "Parasite Temperature control" in "Control" page. There are some simple parameters, like target temperature, triggering temperature, minimum running and idle

You can find some information here: (Sorry, some information must be updated.)
http://vito.tw/poormans-glycol-system/

In my setup, I installed 3 relays, one for Pump to cool beer, one for glycol cooling, and one for pressure control. That's the maximum utilization I can get from a ESP8266.

MrDAnger82 commented:
> В моей установке я установил 3 реле, одно для насоса для охлаждения пива, одно для охлаждения гликоля и одно для контроля давления. Это максимальное использование, которое я могу получить от ESP8266.

Yes, this is what I would like to do ...
But how to measure the temperature of glycol? To put the DS1820 sensor?

vitotai commented:
Assign a DS18B20 as "Room" temperature sensor. Put it in the glycol.

To use BPL with glycol cooling. Don't assign "Fridge" sensor. so that BPL will get "fridge" temperature from "beer" sensor. That's how it works.

MrDAnger82 commented:
Thank. This is what I needed to find out.
I will try.

MrDAnger82 commented:
Vito, please specify.
I connect the sensor and set it on the “ROOM Temp” setup, on the PTC Cooler glycol cooling relay, on the relay turning on glycol circulation in the fermenter’s jacket, “Chamber Cooler”, and on iSpindel- “beer temp”? It is right?

vitotai commented:
Let me clarify. You want to
-use iSpindel as beer sensor
=> set “External Sensor” as “Beer Sensor”
-chill beer by pumping glycol
=> Don’t assign Fridge sensor. Set the actuator controlling PUMP as Chamber cooling. Set minimum idle/cooling time to smaller values.
-control temperature of glycol
=> Assign the sensor in glycol “Room” sensor. Assign the actuator “PTC cooler”.

MrDAnger82 commented:
Vito, I can’t set up cooling.
Please tell me, please, how to adjust the stabilization of the BEER temperature by circulating glycol in the tank jacket using the iSpindel sensor?
The glycol chiller will be autonomous for now.
And then I already tried everything that is possible.
I set the PID parameters and on the page testcmd.htm also configured everything, but it does not work ...
I have beer brewing tomorrow or the day after tomorrow, but I can’t automate anything ...

MrDAnger82 commented:
Here are the screenshots.
![Screenshot_2019-09-09-19-03-27](https://user-images.githubusercontent.com/54216845/64590261-d226b380-d3af-11e9-9746-92055934aad1.png)
![Screenshot_2019-09-09-19-03-11](https://user-images.githubusercontent.com/54216845/64590265-d5ba3a80-d3af-11e9-9adc-1a49ef6ddbbe.png)
![Screenshot_2019-09-09-19-03-00](https://user-images.githubusercontent.com/54216845/64590271-d8b52b00-d3af-11e9-99c9-5a6e5bcf1f18.png)
![Screenshot_2019-09-09-09-13-49](https://user-images.githubusercontent.com/54216845/64590276-db178500-d3af-11e9-8c9c-9a04824260bc.png)

vitotai commented:
Your "Device Setup" seems right to me.
However, the two pages doesn't agree to each other. One is set to "Beer Constant" while the other is "Fridge Constant". I don't see how and why it doesn't work.

The report period of iSpindel should be less than 3 minutes. One minute or even smaller might be better. you need to try to find a value that works best.

"minCoolTime" and "minCoolIdleTime" should be changed to smaller values, like 10 seconds. 
Also change "coolTargetL" and "idleRangeH" to small values, like 0.1, so that the cooling will starts/stops within 0.1 ranges.

However, expect overshooting since the temperature readings change only when iSpindel reports. For example, if you set 1 minute report period and beer temp to 5, cooling starts when the temperature is higher than 5.1. For BPL, the temperature stays at 5.1 until next report. The beer will be cooled for 1 minutes. If the glycol temperature is very low and heat exchange are very efficiency, the real beer temperature might be far lower than expected.

MrDAnger82 commented:
Thanks, I'll try.
As for screenshots with fridge const and beer const, I just tried different settings.

MrDAnger82 commented:
Now I have managed to launch a beer temperature control from iSpindel.
But the chiller ON relay does not respond to the DS18b20 sensor.

MrDAnger82 commented:
How I managed to start the temperature control of beer from iSpindel I did not understand. It seems that he simply erased the flash and “filled” everything again.
The only thing that changed is the device slots ...
Can they somehow be interconnected?

MrDAnger82 commented:
![слоты](https://user-images.githubusercontent.com/54216845/64607323-f9da4380-d3d0-11e9-91ce-705978ad8533.png)

MrDAnger82 commented:
But the last two do not work ... (in the screenshot above)
There is no “connection” between the temperature sensor ROOM (glycol) D6 and the relay for controlling the chiller D8 ...

vitotai commented:
According to the pictures, the temperature is within the range, so the PTC/D8 is not kicked. You can try pull out the sensor to make the temperature rise over the range.

MrDAnger82 commented:
I took out the sensor, it did not help.
The processor also sometimes freezes, shows a temperature increase from iSpindel, but there is no reaction of the pump relay ... I have to reboot and it works again in normal mode.
Also, after a reboot, iSpindel displays a temperature of 73 ° C, gradually it drops and goes to the correct one. iSpindel can’t be taken out of the refrigerator.

vitotai commented:
I have used PTC for 1+ years, and it has worked great for me.

Combined your settings and system freezing problem, you should check your D8 wiring.

D8 is a special PIN that should be LOW so that the system can boot up, otherwise the system will freeze when crash.

I would suggest to try removing wiring from D8 and using D7 as PTC Cooler.

BTW, in my experiences, the system is very unstable and resets from time to time if "nonterrup()/interrupt()" are called frequently. Given the fact that on board ADC can't be stable enough, I decided to remove the "noInterrupt()/interrupt()" calls.

MrDAnger82 commented:
But, D7 controls pressure for me.

MrDAnger82 commented:
On D4, can I transfer pressure control without altering the code?

MrDAnger82 commented:
No, can I transfer the control of the glycol chiller relay to D4? ..

vitotai commented:
D4 is also a special PIN which must be HIGH to enable normal boot up. It is used as button input and is good to be HIGH. You must change code to use D4.
Even you mange to use D4, you might still have problem with wiring.

MrDAnger82 commented:
And then what pin do you recommend connecting devices to?
I don’t use buttons, only WEB, as well as I don’t use the LCD screen.

vitotai commented:
You won't solve problem by switching to D4 or anything else.

Your system will freeze from time to time or your relay will not work if you don't solve wiring problem.

I suggest you to change to D7 is to DEBUG/verify the relay or PINs. If the wiring is OK. D8 should work like D7. At least they do on my setup.

The wiring problem can't be solved buy any change of software, and can't be solved by me.

lbussy commented:
> BTW, in my experiences, the system is very unstable and resets from time to time if "nonterrup()/interrupt()" are called frequently. Given the fact that on board ADC can't be stable enough, I decided to remove the "noInterrupt()/interrupt()" calls.

Do you have a reference for that or is it something you just learned the hard way?

vitotai commented:
> Got ya. Sometimes when a small change makes things better, you end up not really worrying about why. I have a few of those things myself. :)

I guess it has something to do with WiFi, which we don't have source code access. Even I did find out the root cause, the solution will be still a work-around.

MrDAnger82 commented:
Vito, my Gravity switch does not turn off the purple graph, I can’t look at the normally filtered density graph ...
And, yes, you're right, I disconnected the wire from the D8 until the processor stopped hanging, everything works. The chiller itself still maintains the glycol temperature, without BPL intervention.

MrDAnger82 commented:
Vito, I'm sorry, but I can’t figure out the beer profile.
I need to keep the temperature 12C from SG 1.055 to reach 1.020. From 1.020 to achieve stable SG (calculated 1.013), it is necessary to maintain a temperature of 16C.
With the achievement of SG 1.013, proceed to cooling to 0-1C for 4 days.
Did I schedule correctly? It bothers you that you have to set days everywhere, but reaching SG can take more days.
Can you give an example of what a beer profile should look like according to my plan described above?
![бирпрофиль](https://user-images.githubusercontent.com/54216845/64904326-06141880-d6d1-11e9-9a00-bfc0a806baa2.png)

MrDAnger82 commented:
Also, there is no pressure graph in the logging process.

vitotai commented:
First of all, the gravity based beer profile is very complicated. It's also very difficult to debug. I have thought of a way to simulate and debug, but it takes some time to do and not yet done.

You can find some information here:
https://github.com/vitotai/BrewPiLess/blob/master/doc/BeerProfile.md

Assuming OG is 1.055, 
1. 12C from SG 1.055 to reach 1.020
2. 16C from 1.020 to achieve stable SG (calculated 1.013)
3. 0-1C for 4 days

You need insert RAMPs between temperature. If you don't care how long it takes, just give a small value, like 0.1( 2.4hours). So you will have
1. 12C from SG 1.055 to reach 1.020
1.5 Ramp 0.1 day
2. 16C from 1.020 to achieve stable SG (calculated 1.013)
2.5 Ramp 0.1 day
3. 0-1C for 4 days

The days specified in 1 and 2 are used to display the chart. They are also used to speculate current stages if the system reset. I would suggest to give a reasonable values.


![螢幕快照 2019-09-16 上午9 56 34](https://user-images.githubusercontent.com/15953213/65312420-0cb8f900-dbc5-11e9-9980-8b03b08b7b55.jpg)

Regarding the pressure chart, I just don't get it. You show some graph but now you say you don't.

vitotai commented:
Some update:
1. I tried a few pins and found DS18B20 doesn't work on some PINs, like 25 or 26. I don't know if it has something to do with my board. Another OneWire library is used instead of the "stander" one. I have to verify if it is necessary later. 
2. EEPROM related code update. 
3. Currently working on Rotary Encoder. I need to setup JTAG for debugging a crash problem.

MrDAnger82 commented:
Thank!
I will try.

As for the pressure graph:
I turned on logging for recording, there is no pressure graph on the page under the density and temperature graphs.

MrDAnger82 commented:
![Screenshot 2019-09-28 17 29 26](https://user-images.githubusercontent.com/54216845/65817962-98f69c00-e215-11e9-8756-647e0b57d1df.png)

vitotai commented:
In your browser. Type the following URL (suppose the BPL is at 192.168.0.94)
http://192.168.0.94/chart.php

A file named "chart.php" should be downloaded. It is the current log file. Please provide it so that I can check.

MrDAnger82 commented:
ок
[chart.zip](https://github.com/vitotai/BrewPiLess/files/3693224/chart.zip)

vitotai commented:
There is NO valid pressure reading in the log, so the pressure chart is not shown. I don't know why there isn't any pressure record.

vonfritz committed:
add italian translation

vitotai committed:
Merge pull request #199 from petobena/4.0pre

slovak 4.0

vonfritz committed:
update italian json

vitotai committed:
Temperature unit issue in control page

vitotai committed:
Merge branch '4.0pre' of https://github.com/vitotai/BrewPiLess into remote_sync

vitotai committed:
update Slovak locale

vitotai committed:
add missed css file, fix BWF.ajax issue, and platformio.ini issue

vitotai committed:
fine tune WiFi, for issue failed to provide IP under AP mode

vitotai committed:
Fix classic frontend

vonfritz committed:
Merge remote-tracking branch 'upstream/4.0pre' into 4.0preit

vonfritz committed:
update italian locale

vitotai committed:
Merge pull request #213 from vonfritz/4.0preit

4.0preit

vitotai committed:
add calibration field in setup.htm

:

vitotai committed:
refine WiFi behaviour

vitotai committed:
merge dns resolution

vitotai committed:
device setup calibration bug

vitotai committed:
fix ESP32 rotary PINS

vitotai committed:
clear Device Setting cache fo ESp32

vitotai committed:
fix messy chart generated by syncing time tag

vitotai committed:
add bin files

vitotai committed:
adc_type not included

vitotai committed:
misuse setAutoConnect, should be setAutoReconnect

vitotai committed:
misuse setAutoConnect, should be setAutoReconnect

vitotai committed:
add gain setting for ADS1115

vitotai committed:
fine tune ESP32 ADC; update html/js source

vitotai committed:
add 4.0a9

vitotai committed:
stop using setAutoReconnect()

vitotai committed:
update LogViewer

vitotai committed:
ESP32 system info

vitotai committed:
v.40

vitotai committed:
update Readme.md

vitotai committed:
Store WiFi credential for lost of WiFi issue

vitotai committed:
remove 4.0 binary

vitotai committed:
WiFi issue trial

vitotai committed:
wifi issue trial

vitotai committed:
update document, add ESP32 builing instructions

vitotai committed:
add hostname to remote logging, HTTPS on ESP21

vitotai committed:
add host

vitotai committed:
Merge branch 'v4.1' into Nimble.trial

vitotai committed:
NIMBLE trial

vitotai committed:
chart mouse move error

vitotai committed:
add partitoin.csv

vitotai committed:
build ESP8266

vitotai committed:
update document

vitotai committed:
tilt

vitotai committed:
littleFS

vitotai committed:
add Tilt calibration

vitotai committed:
TILT trial

vitotai committed:
MQTT for SONOFF

keithmycek committed:
Fixed broken link

keithmycek committed:
Fixed compilation errors when debug flag is enabled. Uncommented missing variable declaration in PiLink.

keithmycek committed:
Merge tag 'MyloFiore-Patch1' into develop

Minor doc and compilation bugfixes

keithmycek committed:
Merge branch 'hotfix/MyloFiore-Patch1'

vitotai committed:
Merge pull request #278 from kmycek/develop

Minor doc and compilation bug fixes

KnutHelstad committed:
Added Norwegian language file

vitotai committed:
Merge pull request #283 from KnutHelstad/norwegian

Added Norwegian language file

vitotai committed:
add DHT sensor support

vitotai committed:
revise configuration

vitotai committed:
humidity control

vitotai committed:
Merge branch 'master' of https://github.com/vitotai/BrewPiLess

vitotai committed:
add Norwegian translation

vitotai committed:
add resource files of Norwegian

vitotai committed:
documents

vitotai committed:
add missing file

vitotai committed:
HTML/js error

vitotai committed:
HTML/JS errors

vitotai committed:
extra pressure

vitotai committed:
roll back to framework 2.2.0 for ESP8266

vitotai committed:
2.3.0 seems to work

vitotai committed:
framework 2.6.2

vitotai committed:
change ESP8266 framework to 2.6.2

vitotai committed:
crash on HTTP report

vitotai committed:
add library

vitotai committed:
compiled for BME280

vitotai committed:
add new source'

vitotai committed:
enum type might be 4 bytes

vitotai committed:
add Shared Lcd

vitotai committed:
BME280 seems to work

vitotai committed:
shared LCD

vitotai committed:
update AysncMQTT library

vitotai committed:
add backup

vitotai committed:
change maximum step to 10

vitotai committed:
4.2 alpha 1

vitotai committed:
typo for LCD refresh; automatic enable iSpindel

vitotai committed:
avoid connect MQTT when no wifi

vitotai committed:
add setting sanity check; Missing status words in LCD refresh

vitotai committed:
LCD refresh

vitotai committed:
invalid gravity value from ispindel crashes BPL

vitotai committed:
weird display of update time

vitotai committed:
update time display

vitotai committed:
updating LCD in main loop; MDNS workaround; ADC enhancement

vitotai committed:
rESP32 only:reset LCD timer

MrDAnger82 commented:
Hello Vito.
Hello again from Russia!
There was another problem with autospunding.
One of my acquaintances contacted her, I checked and also found this error.
We noticed that the tongue does not open and does not respond to changes in gravity when specifying the capping control by density.
I have a normally closed valve, I have Pin Type-inverted specified in my device, at a density higher than that specified, it does not open as it should. And it does not respond to changes in density below that specified in the capping control ...
![1](https://user-images.githubusercontent.com/54216845/132129830-903ba3fe-6926-4039-8b3e-50168e6b5d30.png)
![2](https://user-images.githubusercontent.com/54216845/132129833-02398f33-8d6a-483b-8a9c-eeb7953fc8c0.png)
![3](https://user-images.githubusercontent.com/54216845/132129835-f56e116c-6e44-429a-ae8e-1be4d3e24b8c.png)
![4](https://user-images.githubusercontent.com/54216845/132129836-2911be6d-de09-4c7a-8b3f-4809d793d2ca.png)

vitotai committed:
refresh results in scrambled display

vitotai committed:
decouple status line of LCD

vitotai committed:
duplicated hardware list entries

vitotai committed:
document update

vitotai committed:
add humidity to MQTT

vitotai committed:
make humidity sensor support default

vitotai committed:
humidity report

vitotai committed:
revise document

vitotai committed:
update humidity.md

vitotai committed:
update mqtt.md

vitotai committed:
typo

vitotai committed:
Beer Profile invalidated after reset

vitotai committed:
rename document

vitotai committed:
update doc

vitotai committed:
remove SSE option

vitotai committed:
change ESp32 framework to 3.5.0

vitotai committed:
adapt arjenhiemstra's ESPAsyncWebServer for iOS Safari 15

vitotai committed:
upgrade ESP8266 framework to 3.2.0, ArduinoJson to 6.19.3, Change FileSystem to LittleFS

vitotai committed:
ESP32/LittleFS test

vitotai committed:
donot change display in HTTP request processing, which might result in race condition

vitotai committed:
4.2 prerelease 1

vitotai committed:
wrong parameter

vitotai committed:
batery display position adjust

vitotai committed:
formula not updated automatically

vitotai committed:
formula not automatically updated

vitotai committed:
longer timeout for data request

vitotai committed:
filemanager for LittleFS

vitotai committed:
move htmljs directory

vitotai committed:
fix include path for moving htmljs

vitotai committed:
revise profile save/load dialog

vitotai committed:
Beer profile truncate error, seting backup/restore

vitotai committed:
case sensitive for file manager

vitotai committed:
rename doc to docs

vitotai committed:
ren README.md

vitotai committed:
Set theme jekyll-theme-minimal

vitotai committed:
add ESP32 w/ LittleFS binary

vitotai committed:
split SmartDisplay.cpp

vitotai committed:
move status line drawing to BrewPiLcd

vitotai committed:
Dedicated display for iSpindel no OLED LCD

vitotai committed:
no signal icon

vitotai committed:
add binaries

vitotai committed:
Merge branch 'master' of https://github.com/vitotai/BrewPiLess

vitotai committed:
update documents

vitotai committed:
update documents

vitotai committed:
remove extra display() call

vitotai committed:
missing setColor() result in missing fixedparts

vitotai committed:
update bins

vitotai committed:
detail iSpindel information

vitotai committed:
add binary

vitotai committed:
custom char

vitotai committed:
merge SmartDisplay and DisplayDisplay

vitotai committed:
recover docs

vitotai committed:
bineries

vitotai committed:
missed IP

vitotai committed:
make LittleFS default for ESP32

vitotai committed:
missed partition table

vitotai committed:
release candidate 1

vitotai committed:
OneWire library for ESP32

vitotai committed:
ESP32 requires old OneWire library

vitotai committed:
wrong filtering of invalid gravity data when plato is used

vitotai committed:
custom char for WiFi signal

vitotai committed:
4.2rc4

vitotai committed:
upgrade to framework 5.0.0 for ESP32

vitotai committed:
file manager directory name issue

vitotai committed:
file manager directory name issue

vitotai committed:
change LITTLEFS.h to LittleFS.h

vitotai committed:
wrong pressure display

vitotai committed:
wrong pressure display

vitotai committed:
save platform.ini

vitotai committed:
platform.ini

vitotai committed:
check packetid returned from publish(). Add restart url

vitotai committed:
missing push to psi array results in failure at exporting

vitotai committed:
update html/js file for wrong build commmand

havardAasen committed:
Flush data to flash on ESP8266 and ESP32

When we write logs, we need to flush the data to flash, so we don't lose
it. This was not possible on ESP8266 because of definitions to only allow
ESP32 access to these functions. This restriction is now removed.

vitotai committed:
negative plato value

vitotai committed:
update Dygraph library, display gravity changes instead of stability

vitotai committed:
revise gravity change calculation

vitotai committed:
using filtered in gravity change calculation

vitotai committed:
release 4.2

vitotai committed:
MQTT spudning control issue

vitotai committed:
remove minimum cooling/warming time option

vitotai committed:
report RSSI to MQTT

vitotai committed:
save

vitotai committed:
checking disconnection of sensor

vitotai committed:
platform.ini

vitotai committed:
4.2.2, no glycol support by default

vitotai committed:
typo in lcd page

vitotai committed:
update framework to 5.4.0 for ESP32, 4.2.0 for ESP8266

vitotai committed:
advance version to 4.3

vitotai committed:
fix logviewer. add new calibration points to logs

vitotai committed:
update HTML

vitotai committed:
make glycol cooling an option in system config

vitotai committed:
filemanger with dir issue

vitotai committed:
remove glycol build

vitotai committed:
update docs

vitotai committed:
split BLE specific functions

vitotai committed:
tilt done

vitotai committed:
add Pill

vitotai committed:
compiled

vitotai committed:
tilt seems done

vitotai committed:
seems to work, but need to fine tune BLE

vitotai committed:
save before branch

vitotai committed:
scan forever

vitotai committed:
gravity informatoin report

vitotai committed:
revise brewPi interface

vitotai committed:
update HTML & simple doc for Pill

vitotai committed:
update HTML/JS

vitotai committed:
Merge pull request #393 from havardAasen/logging

Flush data to flash on ESP8266 and ESP32

vitotai committed:
make ESP8266 compiled

vitotai committed:
delay longer to get NTP response

vitotai committed:
Bugfixed: wrong function called to determine time

vitotai committed:
perentage battery for Pill

vitotai committed:
saved data points

vitotai committed:
rename periodial report function

vitotai committed:
wrong order

vitotai committed:
resume errors

vitotai committed:
gravity report

vitotai committed:
integer value for settings

vitotai committed:
add wireless hydrometer information in log

vitotai committed:
add wireless hydrometer name to MQTT

vitotai committed:
the sequence of CC constants and beer profile might not be in order

vitotai committed:
ESP8266 gravity page

vitotai committed:
revise rx level bar, display the number of real reception

vitotai committed:
temperature correction wrong for Fahrenheit

vitotai committed:
add errorcode

vitotai committed:
split errorcode to another file

vitotai committed:
fix v7 invalid for logviewer version check

vitotai committed:
revise error code

vitotai committed:
beer profile start date conversion

vitotai committed:
PTC and logviewer for V5 log

vitotai committed:
wireless temperature sensor initialization issue

vitotai committed:
sort calibration points

vitotai committed:
1.calibration float value, 2. room humidity sensor & sensor switch issue

vitotai committed:
logviewer exports wrong calibration data

vitotai committed:
wrong tag for room humidity and check validness

vitotai committed:
change esp32 framework backto 6.3.2 and fix typo

vitotai committed:
advance NimBLE to 1.4.1

vitotai committed:
scan/ found

vitotai committed:
update HTML/JS sources

vitotai committed:
add BTHomeListener

vitotai committed:
humidity works

vitotai committed:
temperature sensor

vitotai committed:
support BLE thermometer & humidiy sensor

vitotai committed:
AP password setting

Commit:
feat: add esp32 Sonoff devices support

zandegran committed:
Update 12.HardwareSONOFF.md

zandegran committed:
Update platformio.ini

vitotai committed:
Merge pull request #438 from zandegran/esp32_sonoff_oled

feat: add esp32 Sonoff devices support

Commit:
Humidity control fixed

Commit:
rename member functions of BleScanner

Commit:
RAPT Bluetooth thermometer support

Commit:
active scan

Commit:
upgrade to NIMBLE 2.x

Commit:
Temp correction

Commit:
wrong length of Humidity fixed text

Commit:
use double for flat temperature

Commit:
use double for temperature

Commit:
directives for ESP8266
